packages/alfa-rules/src/sia-r15/rule.ts

```diff
@@ -61,11 +63,23 @@ export default Rule.Atomic.of<Page, Iterable<Element>, Question>({
             }, Map.empty<Option<string>, List<Element>>())
             .values()
         );
+        return filter(groups, (group) => group.size > 1);
       },
 
       expectations(target) {
+        // @see https://html.spec.whatwg.org/multipage/iframe-embed-object.html#process-the-iframe-attributes
         const sources = Set.from(
-          map(target, (iframe) => iframe.attribute("src"))
+          map(target, (iframe) =>
+            iframe
+              .attribute("srcdoc")
+              .map((srcdoc) => srcdoc.value)
+              .getOr(
+                iframe
+                  .attribute("src")
+                  .map((source) => source.value)
+                  .getOr("about:blank")
+              )
+          )
```

Comment:
For `<iframe>` elements without a `src` attribute, we should either do a structural comparison to determine content document equality, or simply always pose a question when there's more than one such iframe.